client/src/components/App/Content/LandingPage/InvestigationTable/InvestigationTableFooter/GroupedInvestigations/GroupedInvestigationsForm/GroupedInvestigationsForm.tsx

```diff
@@ -0,0 +1,127 @@
+import { Autocomplete } from '@material-ui/lab';
+import { yupResolver } from '@hookform/resolvers';
+import { Controller, useForm } from 'react-hook-form';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Grid, TextField, CircularProgress, Collapse } from '@material-ui/core';
+
+import FormInput from 'commons/FormInput/FormInput';
+
+import useStyles from './GroupedInvestigationsFormStyles';
+import { get } from 'Utils/auxiliaryFunctions/auxiliaryFunctions';
+import validationSchema, { OTHER } from './GroupedInvestigationsSchema';
+import GroupedInvestigationsFields from './GroupedInvestigationsFields';
+
+import useGroupedInvestigationsForm, { Reason } from './useGroupedInvestigationsForm';
+
+const GroupedInvestigationsForm : React.FC<Props> = () => {
+
+    const [isReasonsAutoCompleteOpen, setIsReasonsAutoCompleteOpen] = useState<boolean>(false);
+    const [reasons, setReasons] = useState<Reason[]>([]);
+
+    const { fetchReasons } = useGroupedInvestigationsForm({ setReasons });
+
+    const reasonsAutoCompleteLoading = useMemo(() => {
+        return isReasonsAutoCompleteOpen && reasons.length === 0
```

Comment:
why checking the boolean when you can check only the array?

Reply:
beacuse i want to display the CircularProgress only when its open.
I don't want it to be on the DOM when the autoCompltete is close